File: Ent_UsAdmin/BuscarSesion.vb
Imports System.ComponentModel
Imports System.Data.SqlClient

Imports MySql.Data.MySqlClient
Public Class BuscarSesion

    Dim dataConsulta As DataTable
    Dim adapterConsulta As MySqlDataAdapter
    Dim consulta As String
    Dim consultar As Boolean
    Private Sub BackgroundWorker1_DoWork(sender As Object, e As System.ComponentModel.DoWorkEventArgs) Handles BackgroundWorker1.DoWork
        Dim conexionSesiones As MySqlConnection
        conexionSesiones = New MySqlConnection()
        conexionSesiones.ConnectionString = "server=www.prestamosconfia.com;user id=ajas;pwd=123456;port=3306;database=USRS;Convert Zero Datetime=True"
        conexionSesiones.Open()
        adapterConsulta = New MySqlDataAdapter(consulta, conexionSesiones)
        dataConsulta = New DataTable
        adapterConsulta.Fill(dataConsulta)
    End Sub

    Private Sub BuscarClienteSolicitud_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        consulta = "select id,Usuario as Usuario,Empresa,UltimoAcceso,Sistema  from Sesiones where Activo = 1"
        Cargando.Show()
        Cargando.MonoFlat_Label1.Text = "Consultando"
        BackgroundWorker1.RunWorkerAsync()
    End Sub

    Private Sub BackgroundWorker1_RunWorkerCompleted(sender As Object, e As RunWorkerCompletedEventArgs) Handles BackgroundWorker1.RunWorkerCompleted
        dtdatos.DataSource = dataConsulta
        consultar = False
        Cargando.Close()
    End Sub



    Private Sub txtnombre_KeyDown(sender As Object, e As KeyEventArgs) Handles txtnombre.KeyDown
        If e.KeyCode = Keys.Enter Then
            If consultar Then
                MessageBox.Show("Consultando espere...")
            Else
                consulta = "select id,Usuario as Usuario,Empresa,UltimoAcceso,Sistema  from Sesiones where Activo = 1 and usuario like '%" & txtnombre.Text & "%'"
                consultar = True
                Cargando.Show()
                Cargando.MonoFlat_Label1.Text = "Consultando"
                If BackgroundWorker1.IsBusy = True Then
                Else
                    BackgroundWorker1.RunWorkerAsync()
                End If

            End If

        End If
    End Sub

    Private Sub Panel1_Paint(sender As Object, e As PaintEventArgs) Handles Panel1.Paint

    End Sub

    Private Sub Panel1_MouseDown(sender As Object, e As MouseEventArgs) Handles Panel1.MouseDown
        If e.Button = MouseButtons.Left Then
            MoveForm(Me)
        End If
    End Sub

    Private Sub txtnombre_OnValueChanged(sender As Object, e As EventArgs) Handles txtnombre.OnValueChanged

    End Sub

    Private Sub dtdatos_CellContentClick(sender As Object, e As DataGridViewCellEventArgs) Handles dtdatos.CellContentClick

    End Sub

    Private Sub dtdatos_CellDoubleClick(sender As Object, e As DataGridViewCellEventArgs) Handles dtdatos.CellDoubleClick
        CrearNotificacion.txtIdSesion.Text = dtdatos.CurrentRow.Cells(0).Value
        CrearNotificacion.txtUsuario.Text = dtdatos.CurrentRow.Cells(1).Value

        Me.Close()
    End Sub

    Private Sub dtdatos_KeyDown(sender As Object, e As KeyEventArgs) Handles dtdatos.KeyDown
        If e.KeyCode = Keys.Enter Then
            CrearNotificacion.txtIdSesion.Text = dtdatos.CurrentRow.Cells(0).Value
            CrearNotificacion.txtUsuario.Text = dtdatos.CurrentRow.Cells(1).Value

            Me.Close()
        End If
    End Sub
End Class

File: Ent_UsAdmin/Acceso/BuscarSesion.cs
using System;
using System.ComponentModel;
using System.Data;
using System.Windows.Forms;
using Microsoft.VisualBasic.CompilerServices;

using MySql.Data.MySqlClient;

namespace ConfiaAdmin
{
    public partial class BuscarSesion
    {

        private DataTable dataConsulta;
        private MySqlDataAdapter adapterConsulta;
        private string consulta;
        private bool consultar;

        public BuscarSesion()
        {
            InitializeComponent();
        }
        private void BackgroundWorker1_DoWork(object sender, DoWorkEventArgs e)
        {
            MySqlConnection conexionSesiones;
            conexionSesiones = new MySqlConnection();
            conexionSesiones.ConnectionString = "server=www.prestamosconfia.com;user id=ajas;pwd=123456;port=3306;database=USRS;Convert Zero Datetime=True";
            conexionSesiones.Open();
            adapterConsulta = new MySqlDataAdapter(consulta, conexionSesiones);
            dataConsulta = new DataTable();
            adapterConsulta.Fill(dataConsulta);
        }

        private void BuscarClienteSolicitud_Load(object sender, EventArgs e)
        {
            consulta = "select id,Usuario as Usuario,Empresa,UltimoAcceso,Sistema  from Sesiones where Activo = 1";
            My.MyProject.Forms.Cargando.Show();
            My.MyProject.Forms.Cargando.MonoFlat_Label1.Text = "Consultando";
            BackgroundWorker1.RunWorkerAsync();
        }

        private void BackgroundWorker1_RunWorkerCompleted(object sender, RunWorkerCompletedEventArgs e)
        {
            dtdatos.DataSource = dataConsulta;
            consultar = false;
            My.MyProject.Forms.Cargando.Close();
        }



        private void txtnombre_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Enter)
            {
                if (consultar)
                {
                    MessageBox.Show("Consultando espere...");
                }
                else
                {
                    consulta = "select id,Usuario as Usuario,Empresa,UltimoAcceso,Sistema  from Sesiones where Activo = 1 and usuario like '%" + txtnombre.Text + "%'";
                    consultar = true;
                    My.MyProject.Forms.Cargando.Show();
                    My.MyProject.Forms.Cargando.MonoFlat_Label1.Text = "Consultando";
                    if (BackgroundWorker1.IsBusy == true)
                    {
                    }
                    else
                    {
                        BackgroundWorker1.RunWorkerAsync();
                    }

                }

            }
        }

        private void Panel1_Paint(object sender, PaintEventArgs e)
        {

        }

        private void Panel1_MouseDown(object sender, MouseEventArgs e)
        {
            if (e.Button == MouseButtons.Left)
            {
                Module_resize.MoveForm(this);
            }
        }

        private void txtnombre_OnValueChanged(object sender, EventArgs e)
        {

        }

        private void dtdatos_CellContentClick(object sender, DataGridViewCellEventArgs e)
        {

        }

        private void dtdatos_CellDoubleClick(object sender, DataGridViewCellEventArgs e)
        {
            My.MyProject.Forms.CrearNotificacion.txtIdSesion.Text = Conversions.ToString(dtdatos.CurrentRow.Cells[0].Value);
            My.MyProject.Forms.CrearNotificacion.txtUsuario.Text = Conversions.ToString(dtdatos.CurrentRow.Cells[1].Value);

            Close();
        }

        private void dtdatos_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Enter)
            {
                My.MyProject.Forms.CrearNotificacion.txtIdSesion.Text = Conversions.ToString(dtdatos.CurrentRow.Cells[0].Value);
                My.MyProject.Forms.CrearNotificacion.txtUsuario.Text = Conversions.ToString(dtdatos.CurrentRow.Cells[1].Value);

                Close();
            }
        }
    }
}